server/src/test/integration/org/apache/cassandra/sidecar/acl/RoleBasedAuthorizationIntegrationTest.java

```diff
@@ -0,0 +1,428 @@
+    {
+        Session session = maybeGetSession();
+        session.execute("INSERT INTO system_auth.identity_to_role (identity, role) VALUES (\'" + identity + "\',\'" + role + "\');");
+    }
+
+    private void createKeyspace(String keyspace)
+    {
+        Session session = maybeGetSession();
+        session.execute("CREATE KEYSPACE IF NOT EXISTS " + keyspace + " WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':'3'}");
+    }
+
+    private void createTable(String keyspace, String table)
+    {
+        Session session = maybeGetSession();
+        session.execute(String.format("CREATE TABLE %s.%s (a int, b text, PRIMARY KEY (a));", keyspace, table));
+        session.execute("INSERT INTO " + keyspace + "." + table + " (a, b) VALUES (1, 'text');");
+    }
+
+    private void grantKeyspacePermission(String keyspace, String role)
+    {
+        Session session = maybeGetSession();
+        session.execute("GRANT ALL PERMISSIONS ON KEYSPACE " + keyspace + " TO " + role);
+    }
+
+    private void grantTablePermission(String keyspace, String table, String role)
+    {
+        Session session = maybeGetSession();
+        session.execute("GRANT ALL PERMISSIONS ON " + keyspace + "." + table + " TO " + role);
+    }
+
+    private void grantSidecarPermission(String role, String resource, String permission)
+    {
+        Session session = maybeGetSession();
+        session.execute(String.format("INSERT INTO sidecar_internal.role_permissions_v1 (role, resource, permissions) " +
+                                      "VALUES ('%s', '%s', {'%s'})", role, resource, permission));
+    }
+
+    private void updateSidecarPermission(String role, String resource, String permission)
+    {
+        Session session = maybeGetSession();
+        session.execute(String.format("UPDATE sidecar_internal.role_permissions_v1 SET permissions = permissions + {'%s'} " +
+                                      "where role = '%s' and resource = '%s'", permission, role, resource));
+    }
+
+    private SslConfiguration sslConfigWithKeystoreTruststore()
+    {
+        return SslConfigurationImpl
+               .builder()
+               .enabled(true)
+               .keystore(new KeyStoreConfigurationImpl(clientKeystorePath.toAbsolutePath().toString(), clientKeystorePassword, "PKCS12"))
+               .truststore(new KeyStoreConfigurationImpl(truststorePath.toAbsolutePath().toString(), truststorePassword, "PKCS12"))
+               .build();
+    }
+
+    private void verifyAccess(VertxTestContext context, HttpMethod method, String testRoute, Path clientKeystorePath)
+    {
+        Checkpoint checkpoint = context.checkpoint();
+        verifyAccess(context, checkpoint, method, testRoute, clientKeystorePath, false);
+    }
+
+    private void verifyAccess(VertxTestContext context, Checkpoint checkpoint, HttpMethod method,
+                              String testRoute, Path clientKeystorePath, boolean expectForbidden)
```

Comment:
Nit: Please consider changing the flag meaning to a `expectAllowed`. I know this may be personal preference, but in my head, passing a true "should" mean that the access should be allowed.

Reply:
We use this method to check both access allowed cases and access denied cases. Hence named it `verifyAccess`